### PR TITLE
core: Remove Server class

### LINKED ISSUE #412: Get rid of `Server` in favour of `app.start()`
See [wiki](https://github.com/strongloop/loopback-next/wiki/Hello-world):

```ts
import {Application} from '@loopback/core';

const app = new Application();
app.handler((sequence, request, response) => {
  sequence.send(response, 'hello world');
});

(async function start() {
  await app.start();
  console.log(`The app is running on port ${app.getSync('http.port')}`);
})();
```

### PR DESCRIPTION
**Pull request by bajtos**
Remove the Server class, move Server.start to Application.start.

Closes #412

cc @bajtos @raymondfeng @ritch @superkhau

### CODE REVIEW

**Comment by bajtos** [comment]
Cross-posting a question from our internal chat for posterity:

> How are we going to support the following use case?
> 1. Create an api implementation such as Customer.findById
> 2. Expose it to http://localhost:3000, https://localhost:4000, and grpc://localhost:5000

I think the idea is to let the application to control what servers it wants to start (HTTP, HTTPS, etc.), instead of having the user to decide what servers to mount the application on. Later, we can modify `Application.prototype.start()` to start additional servers, depending on the configuration. See #430 for HTTPS.

File: packages/core/test/acceptance/bootstrapping/application.acceptance.ts [hunk 1]
```diff
@@ -111,3 +111,19 @@ describe('Bootstrapping the application', () => {
     });
   });
 });
+
+describe('Starting the application', () => {
+  it('starts HTTP server', async () => {
```
Comment (superkhau) [inline packages/core/test/acceptance/bootstrapping/application.acceptance.ts]: starts a

**Comment by kjdelisle** [comment]
Out of curiosity, do we have any plans to support servers that don't use HTTP protocol?

**Comment by bajtos** [comment]
> you'll probably want to talk it over more with @raymondfeng before merging.

Good point. IIRC, we have already discussed this change with Raymond on Slack several weeks ago, so I'll go ahead with landing this patch.

**Pull request merged**